symengine/printers/unicode.cpp

```diff
@@ -0,0 +1,760 @@
+    // NOTE: Using italics for constants which is very common in mathematics
+    // typesetting. (It goes against the ISO typesetting-standard though.)
+    if (eq(x, *pi)) {
+        box_ = StringBox("\U0001D70B", 1);
+    } else if (eq(x, *E)) {
+        box_ = StringBox("\U0001D452", 1);
+    } else if (eq(x, *EulerGamma)) {
+        box_ = StringBox("\U0001D6FE", 1);
+    } else if (eq(x, *Catalan)) {
+        box_ = StringBox("\U0001D43A", 1);
+    } else if (eq(x, *GoldenRatio)) {
+        box_ = StringBox("\U0001D719", 1);
+    }
+}
+
+StringBox UnicodePrinter::apply(const vec_basic &d)
+{
+    StringBox box;
+    StringBox comma(", ");
+    for (auto p = d.begin(); p != d.end(); p++) {
+        if (p != d.begin()) {
+            box.add_right(comma);
+        }
+        StringBox arg = apply(*p);
+        box.add_right(arg);
+    }
+    return box;
+}
+
+void UnicodePrinter::bvisit(const Abs &x)
+{
+    StringBox box = apply(*x.get_arg());
+    box.enclose_abs();
+    box_ = box;
+}
+
+void UnicodePrinter::bvisit(const Floor &x)
+{
+    StringBox box = apply(*x.get_arg());
+    box.enclose_floor();
+    box_ = box;
+}
+
+void UnicodePrinter::bvisit(const Ceiling &x)
+{
+    StringBox box = apply(*x.get_arg());
+    box.enclose_ceiling();
+    box_ = box;
+}
+
+void UnicodePrinter::bvisit(const Function &x)
+{
+    StringBox box(names_[x.get_type_code()], lengths_[x.get_type_code()]);
+    vec_basic vec = x.get_args();
+    StringBox args = apply(vec);
+    args.enclose_parens();
+    box.add_right(args);
+    box_ = box;
+}
+
+void UnicodePrinter::bvisit(const FunctionSymbol &x)
+{
+    StringBox box(x.get_name());
+    StringBox args;
+    StringBox comma(", ");
+    bool first = true;
+    for (auto arg : x.get_args()) {
+        if (first) {
+            first = false;
+        } else {
+            args.add_right(comma);
+        }
+        StringBox argbox = apply(arg);
+        args.add_right(argbox);
+    }
+    args.enclose_parens();
+    box.add_right(args);
+    box_ = box;
+}
+
+StringBox UnicodePrinter::parenthesizeLT(const RCP<const Basic> &x,
+                                         PrecedenceEnum precedenceEnum)
+{
+    Precedence prec;
+    if (prec.getPrecedence(x) < precedenceEnum) {
+        auto box = apply(x);
+        box.enclose_parens();
+        return box;
+    } else {
+        return apply(x);
+    }
+}
+
+StringBox UnicodePrinter::parenthesizeLE(const RCP<const Basic> &x,
+                                         PrecedenceEnum precedenceEnum)
+{
+    Precedence prec;
+    if (prec.getPrecedence(x) <= precedenceEnum) {
+        auto box = apply(x);
+        box.enclose_parens();
+        return box;
+    } else {
+        return apply(x);
+    }
+}
+
+StringBox UnicodePrinter::apply(const RCP<const Basic> &b)
+{
+    b->accept(*this);
+    return box_;
+}
+
+StringBox UnicodePrinter::apply(const Basic &b)
+{
+    b.accept(*this);
+    return box_;
+}
+
+std::vector<std::string> init_unicode_printer_names()
```

Comment:
maybe start with the default values from [init_str_printer_names()](https://github.com/symengine/symengine/blob/master/symengine/printers/strprinter.cpp#L1038) then set the ones that differ? reduces duplication & you also support new functions automatically when they are added to the main printer

Reply:
Good idea! I'll change this.